spec/ot-date.spec.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+describe("OtDate", () => {
+    var OtDate, MomentFake;
+
+    beforeEach(() => {
+        class MomentFake {
+            constructor() {
+                [
+                    "add",
+                    "fromString",
+                    "toDate",
+                    "utc",
+                    "unix"
+                ].forEach((method) => {
+                    this[method] = jasmine.createSpy(method);
+                    this[method].andCallFake(() => {
+                        return this;
+                    });
+                });
+                this.isBefore = jasmine.createSpy("isBefore");
+                this.isBefore.andCallFake((date) => {
+                    return true;
+                });
+                this.toString = jasmine.createSpy("toString");
+                this.toString.andCallFake(() => {
+                    return "a formatted date";
+                });
+            }
+        };
+        OtDate = require("../lib/ot-date")(MomentFake);
+    });
+    describe("fromDate()", () => {
+        it("gets an OtDate object back from a custom date/time", () => {
+            expect(OtDate.fromDate("2016-04-16")).toEqual(jasmine.any(OtDate));
```

Comment:
There is no check to make sure the date value inside is accurate.

I was thinking about this and I think that the fromDate compliments toDate.  You might want to unmock moment and then organize tests like so:

```
it("goes to and from a date", () => {
    var d, expected;

    d = new Date();
    expected = d.toString();
    ot = OtDate.fromDate(d);
    expect(ot.toDate().toString()).toEqual(expected);
})
```

This particular example is especially bad because `.fromDate()` should operate on a `Date`, not a string.